spec/requests/cinema_halls_controller_spec.rb

```diff
@@ -0,0 +1,110 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+require 'shared_examples_for_cinema_halls'
+
+RSpec.describe CinemaHallsController, type: :request do
+  let(:role) { create(:user_role) }
+  let(:user) { create(:user, user_role: role) }
+
+  describe '#index' do
+    subject { get '/cinema_halls' }
```

Comment:
Subjects might have a descriptive name, like: `subject(:get_cinema_halls) { .. }` and later on in the test, you can use that name instead of `subject` which might be quite cryptic in larger files.

Reply:
Thanks, I have refactored subjects 👍